server/src/main/java/io/druid/server/http/LookupCoordinatorResource.java

```diff
@@ -0,0 +1,226 @@
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+/**
+ * Contains information about lookups exposed through the coordinator
+ */
+@Path("/druid/coordinator/v1/lookups")
+public class LookupCoordinatorResource
+{
+  private static final Logger LOG = new Logger(LookupCoordinatorResource.class);
+  private final LookupCoordinatorManager lookupCoordinatorManager;
+  private final @Smile ObjectMapper smileMapper;
+  private final @Json ObjectMapper jsonMapper;
+
+  @Inject
+  public LookupCoordinatorResource(
+      final LookupCoordinatorManager lookupCoordinatorManager,
+      final @Smile ObjectMapper smileMapper,
+      final @Json ObjectMapper jsonMapper
+  )
+  {
+    this.smileMapper = smileMapper;
+    this.jsonMapper = jsonMapper;
+    this.lookupCoordinatorManager = lookupCoordinatorManager;
+  }
+
+  @GET
+  @Produces({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  public Response getTiers()
+  {
+    try {
+      final Map<String, Map<String, Map<String, Object>>> knownNamespaces = lookupCoordinatorManager.getKnownLookups();
+      if (knownNamespaces == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      } else {
+        return Response.ok().entity(knownNamespaces.keySet()).build();
+      }
+    }
+    catch (Exception e) {
+      LOG.error(e, "Error getting list of lookups");
+      return Response.serverError().entity(ServletResourceUtils.sanitizeException(e)).build();
+    }
+  }
+
+  @POST
+  @Produces({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  @Consumes({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  public Response updateAllNamespaces(
+      InputStream in,
+      @HeaderParam(AuditManager.X_DRUID_AUTHOR) @DefaultValue("") final String author,
+      @HeaderParam(AuditManager.X_DRUID_COMMENT) @DefaultValue("") final String comment,
+      @Context HttpServletRequest req
+  )
+  {
+    try {
+      final boolean isSmile = SmileMediaTypes.APPLICATION_JACKSON_SMILE.equals(req.getContentType());
+      final ObjectMapper mapper = isSmile ? smileMapper : jsonMapper;
+      final  Map<String, Map<String, Map<String, Object>>> map;
+      try {
+          map = mapper.readValue(in, new TypeReference<Map<String, Map<String, Map<String, Object>>>>()
+          {
+          });
+      } catch (IOException e) {
+        return Response.status(Response.Status.BAD_REQUEST).entity(ServletResourceUtils.sanitizeException(e)).build();
+      }
+      lookupCoordinatorManager.updateLookups(map, new AuditInfo(author, comment, req.getRemoteAddr()));
+      return Response.status(Response.Status.ACCEPTED).entity(map).build();
+    }
+    catch (Exception e) {
+      LOG.error(e, "Error creating new lookups");
+      return Response.serverError().entity(ServletResourceUtils.sanitizeException(e)).build();
```

Comment:
The current impl sends all lookups to a host on a config change and lets the host figure out what to do about them.

I was considering making the "get new host" logic use the zk node version number, so that way hosts that disconnect between updates can be guaranteed to get a fresh POST of all the lookups.